contracts/eos/BancorConverterRegistry/BancorConverterRegistry.cpp

```diff
@@ -0,0 +1,165 @@
+    const extended_symbol pool_token = get_converter_pool_token(converter_account, pool_token_code);
+    const converter_t converter = converter_t{ converter_account, pool_token };
+    check(is_converter_active(converter), "converter is inactive");
+
+    register_converter(converter);
+    register_pool_token(converter.pool_token);
+    register_convertible_pairs(converter);
+
+    BancorConverter::reserves reserves_table(converter.contract, converter.pool_token.get_symbol().code().raw());
+    uint64_t num_of_reserves = std::distance(reserves_table.cbegin(), reserves_table.cend());
+    if (num_of_reserves > 1)
+        register_liquidity_pool(converter);
+}
+
+ACTION BancorConverterRegistry::rmconverter(const name& converter_account, const symbol_code& pool_token_code) {
+    const extended_symbol pool_token = get_converter_pool_token(converter_account, pool_token_code);
+    const converter_t converter = converter_t{ converter_account, pool_token };
+    check(has_auth(get_self()) || !is_converter_active(converter), "cannot remove an active converter");
+
+    unregister_converter(converter);
+    unregister_pool_token(converter.pool_token);
+    unregister_convertible_pairs(converter);
+    unregister_liquidity_pool(converter);
+}
+
+void BancorConverterRegistry::register_converter(const converter_t& converter) {
+    converters converters_table(get_self(), converter.pool_token.get_symbol().code().raw());
+    const auto p_converter = converters_table.find(converter.pool_token.get_contract().value);
+    check(p_converter == converters_table.end(), "converter already exists");
+    converters_table.emplace(get_self(), [&converter](converter_t& c){
+        c = converter;
+    });
+}
+
+void BancorConverterRegistry::register_liquidity_pool(const converter_t& liquidity_pool) {
+    liquidity_pools liquidity_pools_table(get_self(), liquidity_pool.pool_token.get_symbol().code().raw());
+    const auto p_liquidity_pool = liquidity_pools_table.find(liquidity_pool.pool_token.get_contract().value);
+    check(p_liquidity_pool == liquidity_pools_table.end(), "liquidity pool already exists");
+    liquidity_pools_table.emplace(get_self(), [&liquidity_pool](converter_t& lp){
+        lp = liquidity_pool;
+    });
+}
+
+void BancorConverterRegistry::register_pool_token(const extended_symbol& pool_token) {
+    pool_tokens pool_tokens_table(get_self(), pool_token.get_symbol().code().raw());
+    const auto p_pool_token = pool_tokens_table.find(pool_token.get_contract().value);
+    check(p_pool_token == pool_tokens_table.end(), "pool token already exists");
+    pool_tokens_table.emplace(get_self(), [&pool_token](token_t& p){
+        p.token = pool_token;
+    });
+}
+
+void BancorConverterRegistry::register_convertible_pairs(const converter_t& converter) {
+    BancorConverter::reserves reserves_table(converter.contract, converter.pool_token.get_symbol().code().raw());
```

Comment:
so we keep pool token -> reserve, reserve -> pool token and reserve -> any other reserve?
not actually sure we need the latter, need to think about it

Reply:
reserve -> any other reserve of the same converter... meaning 1 hop away only